core/src/tests.rs

```diff
@@ -158,7 +158,30 @@ async fn validate_eth() {
     let traces = res.data.traces.iter().map(polars_arrow_to_arrow_rs);
 
     for (((blocks, transactions), logs), traces) in blocks.zip(transactions).zip(logs).zip(traces) {
+        // USING PARQUET TO DEBUG, REMOVE BEFORE MERGING
+        //write blocks, transactions, logs, traces to parquet
+        let blocks_file = std::fs::File::create("blocks.parquet").unwrap();
+        let mut blocks_writer = parquet::arrow::ArrowWriter::try_new(blocks_file, blocks.schema(), None).unwrap();
+        blocks_writer.write(&blocks).unwrap();
+        blocks_writer.close().unwrap();
+
+        let transactions_file = std::fs::File::create("transactions.parquet").unwrap();
+        let mut transactions_writer = parquet::arrow::ArrowWriter::try_new(transactions_file, transactions.schema(), None).unwrap();
+        transactions_writer.write(&transactions).unwrap();
+        transactions_writer.close().unwrap();
+
+        let logs_file = std::fs::File::create("logs.parquet").unwrap();
+        let mut logs_writer = parquet::arrow::ArrowWriter::try_new(logs_file, logs.schema(), None).unwrap();
+        logs_writer.write(&logs).unwrap();
+        logs_writer.close().unwrap();
+
+        let traces_file = std::fs::File::create("traces.parquet").unwrap();
+        let mut traces_writer = parquet::arrow::ArrowWriter::try_new(traces_file, traces.schema(), None).unwrap();
+        traces_writer.write(&traces).unwrap();
+        traces_writer.close().unwrap();
+
```

Comment:
Remove this before merging.

Reply:
I'm seeing the stream delivering incomplete data. Not every log data is being delivered for Block 18123130.

Reply:
could be bug in hypersync query implementation.

Can try changing query to this format instead of using JOIN_ALL

```
{
include_all_blocks: true  
transactions: [{}],
logs: [{}],
traces: [{}]
}
```